Payload/PayloadTrait.php

```diff
@@ -37,25 +37,28 @@ public function keyExists(string $key): bool
     /**
      * @return mixed
      */
-    public function getValue(string $key)
+    public function getValue(string $key, bool $throwIfNotExists = false)
```

Comment:
I think the concept why @wachterjohannes introduced the Payload is that the payload is validated. This currently disable this and is a bc break. You will not longer get an error if you accidently write `titttle` instead of `title` or something. So I think if the service doesn't care about the payload validation it should maybe access its values over `->getPayload()['test'] ?? null`. And if we introduce it from bc atleast the default should be here `true` instead of false.

Reply:
i think the `$throwIfNotExists` flag is a useful addition that shouldnt cause any problems. dont have a strong opinion about the default value, as i think there are valid arguments for both sides 🙂 but @alexander-schranz is right, regarding bc `true` would be better

Reply:
@alexander-schranz @nnatter Introduced a `$defaultThrowIfNotExists` on `initializePayloadTrait()` defaulting to `true`. Would that be ok for you?

Reply:
Yes maybe it would be better to go then with `bool $nullIfNotExists = false`, but don't have strong opinion on that. @wachterjohannes what do you think?

Reply:
i am not sure about that - the validation was a big intention to create this trait - so i would at least prefer the `bool $nullIfNotExists = false ` parameter

Reply:
i would not add a `$defaultThrowIfNotExists` because it makes the trait stateful and this state needs to be saved as class property on the class that is using the trait. this means that the trait will behave differently in different classes, also this means that it may cause problems when it is overwritten unintentionally (not very likely although).

furthermore it makes the interface of the trait less intuitive. nullable booleans are hard to grasp in my opion because the parameter name cannot say anything about the `null` case.

finally, i think it is good that you need to pass an additional parameter if you want to ignore an exception. this makes things explicit 🙂

Reply:
@nnatter @alexander-schranz @wachterjohannes fixed